README.md

```diff
@@ -1,25 +1,108 @@
 
-Please only use `bun` package manager.
+## Table of Contents
 
-### setup frontend
+- [Goal](#goal)
+- [Features](#features)
+- [Technology Stack](#technology-stack)
+- [Getting Started](#getting-started)
+- [User Flows](#user-flows)
+- [Future Enhancements](#future-enhancements)
+- [Contributing](#contributing)
+- [License](#license)
 
-```
+## Goal
 
-bun install && cp .env.example .env.local
+The primary goal is to create a decentralized network for community leaders within Refunite's ecosystem, enabling:
 
-```
+1. Onboarding of community leaders to acknowledge their role within the trust network.
+2. Secure role recovery via peer-to-peer social recovery mechanisms when access is lost.
 
-### run local
+The first iteration is a pilot with leaders in Uganda, where leaders can recover their credentials from fellow Ugandan leaders.
 
-```
-bun dev
-```
+## Features
+
+- **Role Management**: Community leaders receive on-chain credentials through the [Hats Protocol](https://www.hatsprotocol.xyz/) to attest to their roles.
+- **Social Recovery**: Leaders who lose access to their credentials can regain access through peer validation.
+- **Decentralized Trust Network**: No central control over the state; managed entirely by leaders themselves.
+- **Scalability**: Designed to support up to 100,000 leaders, grouped by geographic or other predefined subsets.
+
+## Technology Stack
+
+- **Hats protocol**: Powered by [Hats Protocol](https://www.hatsprotocol.xyz/) for role management and governance.
+- **Frontend**: NextJS web application built with Web3 connectivity.
+- **Silk wallet**: [Silk Wallet](https://www.silk.sc/) for Web3 connectivity.
+- **Gnosis Safe**: [Gnosis Safe](https://gnosis-safe.io/) for multi-sig wallet management.
+- **Blockchain**: Blockchain-based for immutable, secure credential management.
+
+## Getting Started
+
+### Prerequisites
+
+- [Node.js](https://nodejs.org/)
+- [Silk Wallet](https://semaphore.pse.dev/)
+
+### Installation
+
+1. Clone the repository:
+
+   ```bash
+   git clone https://github.com/syntro-fi/refunite-network.git
+   cd refunite-network
+   ```
+
+2. Install dependencies:
 
-### build
+   ```bash
+   bun install
+   ```
 
+3. Run the application:
+
+   ```bash
+   bun dev
+   ```
+
+4. Open in your browser at `http://localhost:3000`.
+
+## User Flows
+
+### 1. Role Onboarding
+
+```mermaid
+sequenceDiagram
+    participant Refunite
+    participant Community_Leader
+    Refunite->>Community_Leader: Validates leader role
+    Community_Leader->>Refunite: Provides wallet address
+    Refunite->>Community_Leader: Assigns leadership token
 ```
-bun run build
+
+### 2. Role Recovery
+
+```mermaid
+sequenceDiagram
+    participant Leader_A (Loses Access)
```

Comment:
This is a bit confusing. why do you have empty swim lanes for Leader_A (Loses Access) and Peer_Leader?